### PR TITLE
:tada: STATEMINE Version Zero

### LINKED ISSUE #5622: Congratulations pop-up appeared too soon
Got this on **beta**: 
Triet to buy [this](https://beta.kodadot.xyz/ksm/gallery/15975192-7092505797a30ccf48-NEON%20BABES-NEON_BABE_11-00000011?page=4) RMRK2 NFT.

The moment I clicked on Buy - Confirm the pop-up appeared.
I did not have a chance to sign up the transaction first.  

Screenshot 2023-04-12 at 15 01 12

Video here
---

https://user-images.githubusercontent.com/104380776/231466694-dfa7308f-1a10-43ce-96e5-c41f22d4d67a.mov

### LINKED ISSUE #5881: Adding Statemine into the chain selector
@vikiival is working on this one, once closed statemine can begin

### PR DESCRIPTION
**Pull request by vikiival**
- :truck: rename statemine to stmn
- :truck: rename statemine to stmn
- :tada: stmn in the house
- :bug: statemine not properly defined
- :coffin: old code
- :zap: nicer organization of landing page
- :tada: statemine
- :zap: enable create on stmn
- :wrench: apollo endpoints
- :wastebasket: deduped code via static
- :wrench: statemine endpoints
- :bug: fixed imports for statick
- :bug: stmn was pointing to subqueries
- :zao: isSnek
- :bug: fixed value of ksm
- :bug: very strange symbols
- :construction: WIP minting
- :zap: enable buy on statemine
- :bug: nft bought is too soon, close #5622
- :bug: try/catch/finally
- :zap: buy for statemine
- :zap: uniqes
- :boom: take url by prefix
- :bug: need to pass empty whitelist as undefined
- :sparkles: list
- :mute: logs because i had inconsistent connection
- :sparkles: send
- :broom: organize imports
- :shrug: may burn works
- :zap: disable minting on statemine
- :sprakles: exec mint Statemine
- :bug: did not set metadata
- :bug: expected String got Undefined
- :sparkles: create collection
- :technologist: one v-if for whole landing page

**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [x] Feature
- [x] Refactoring

## Context

Integration of statemine:
- #5879
- Close #5881

## What should you know

We have of course new indexer called [**stick**](https://github.com/kodadot/stick)

We are missing minting (on-chain code is there and I started to rewrite it with new design)

## Screenshot 📸

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

Screenshot 2023-05-11 at 19 53 25

- [x] [I made some on-chain actions](https://statemine.subscan.io/account/Fksmad33PFxhrQXNYPPJozgWrv82zuFLvXK7Rh8m1xQhe98)

Screenshot 2023-05-11 at 22 44 21

### <samp>🤖 Generated by Copilot at baa480b</samp>

Added support for Statemine chain and its features, such as creating and buying NFTs, to the gallery. Updated various components, composables, and static files to handle Statemine transactions and data. Added new components and files for the Statemine create feature. Fixed some minor issues and improved code style and readability.

### <samp>🤖 Generated by Copilot at baa480b</samp>

> _Sing, O Muse, of the valiant deeds of the gallery heroes,_
> _Who ventured to the realm of Statemine, the chain of many tokens,_
> _And there they wrought with skill and craft new collections and NFTs,_
> _Using `CreateCollection` and `CreateNft`, the components of minting._

### CODE REVIEW

File: components/stmn/Create/CreateCollection.vue [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        :label="$t('mint.collection.description.label')"
+        class="w-full mb-4">
+        <NeoInput
+          v-model="description"
+          type="textarea"
+          has-counter
+          maxlength="500"
+          height="10rem" />
+      </NeoField>
+    </form>
```
Comment (roiLeo) [inline components/stmn/Create/CreateCollection.vue:46]: No submit button?

File: components/stmn/Create/utils.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+export const getDeposit = (token: Token, depositInKSM: number): number => {
+  return token === 'KSM' ? depositInKSM : ksmToBsx(depositInKSM)
+}
+
+export const getFeesToken = async (): Promise<Token> => {
+  const { apiInstance } = useApi()
+  const { accountId } = useAuth()
+  const { $consola } = useNuxtApp()
+  const bsxTokenId = '0'
```
Comment (roiLeo) [inline components/stmn/Create/utils.ts:32]: bsx on stmn?

File: components/stmn/input/TokenBalanceInput.vue [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  get decimals() {
+    return this.asset.decimals
+  }
+
+  public checkValidity() {
+    const balanceInputComponent = this.$refs
+      .magicBalanceInput as BasicBalanceInput
+    return balanceInputComponent.checkValidity()
+  }
+}
```
Comment (roiLeo) [inline components/stmn/input/TokenBalanceInput.vue:45]:
```suggestion
<script lang="ts">
import BasicBalanceInput from '@/components/shared/form/BasicBalanceInput.vue'

const vValue = ref()
const magicBalanceInput = ref(null)
defineProps<{
  tokenId: string | number
}>()

const { decimals, unit } = useChain()
```

File: store/assets.ts [hunk 4]
```diff
@@ -41,6 +41,8 @@ export const actions: ActionTree<TokenState, TokenState> = {
       assetList: [chainAssetOf(prefix)],
     }))
 
+    console.log(assetList, 'assetList')
```
Comment (roiLeo) [inline store/assets.ts:44]: 👋

File: pages/stmn/create/item.vue [hunk 5]
```diff
@@ -0,0 +1,7 @@
+<template>
```
Comment (roiLeo) [inline pages/stmn/create/item.vue:1]: who's this?

File: components/stmn/input/OfferBalanceInput.vue [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+  <BasicBalanceInput
+    ref="magicBalanceInput"
+    v-model="vValue"
+    :decimals="decimals"
+    :min="min"
+    :unit="unit"
+    expanded />
+</template>
+
+<script lang="ts">
```
Comment (roiLeo) [inline components/stmn/input/OfferBalanceInput.vue:11]: vue3?

File: components/stmn/Create/utils.ts [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+    // '1' if paying in ksm (on basilisk)
+    // '5' if paying in ksm (on Snek)
+    const tokenId = await getAssetIdByAccount(api, accountId.value)
+
+    return tokenId === bsxTokenId ? 'BSX' : 'KSM'
+  } catch (e) {
+    $consola.log(e)
+  }
+
+  return 'BSX'
```
Comment (roiLeo) [inline components/stmn/Create/utils.ts:47]: 🤔

File: store/assets.ts (hunk 4 above)
Comment (vikiival) [inline store/assets.ts:44]:
```suggestion
```

File: pages/stmn/create/item.vue (hunk 5 above)
Comment (vikiival) [reply pages/stmn/create/item.vue:1]: Placeholder for new create design

**Comment by vikiival** [comment]
@kodadot/internal-dev any ideas why it does not build? 🥺

**Comment by roiLeo** [comment]
> @kodadot/internal-dev any ideas why it does not build? 🥺

maybe comming from your push of `pnpm-lock.yaml` file, have you try to run build command on local? try to clear node_modules and reinstall

File: composables/transaction/mintToken/transactionMintStatemine.ts [hunk 8]
```diff
@@ -0,0 +1,50 @@
+import { ExecuteTransactionParams } from '@/composables/useTransaction'
+import type { ActionMintToken, MintedCollectionBasilisk } from '../types'
+import { constructMeta } from './constructMeta'
+
+export async function execMintStatemine(
```
[the remark this answers is not in the document]
Comment (yangwao) [reply composables/transaction/mintToken/transactionMintStatemine.ts:5]: what's about this 👀

**Comment by yangwao** [comment]
try your luck and update to pnpm8
- https://github.com/kodadot/nft-gallery/pull/5845

Bike Fall

**Comment by roiLeo** [comment]
Build OK on my side

```bash
✔ Generated public dist                                                                                                                                           nitro 12:24:37
ℹ Initializing prerenderer                                                                                                                                        nitro 12:24:37
ℹ Prerendering 47 initial routes with crawler                                                                                                                     nitro 12:24:38
  ├─ / (24ms)                                                                                                                                                     nitro 12:24:38
  ├─ /200 (2ms)                                                                                                                                                   nitro 12:24:38
  ├─ /404 (1ms)                                                                                                                                                   nitro 12:24:38
  ├─ /about (1ms)                                                                                                                                                 nitro 12:24:38
  ├─ /carbonless (1ms)                                                                                                                                            nitro 12:24:38
  ├─ /contribute (1ms)                                                                                                                                            nitro 12:24:38
  ├─ /e2e-login (1ms)                                                                                                                                             nitro 12:24:38
  ├─ /error (2ms)                                                                                                                                                 nitro 12:24:38
  ├─ /esCarbonless (1ms)                                                                                                                                          nitro 12:24:38
  ├─ /esSustainability (4ms)                                                                                                                                      nitro 12:24:38
  ├─ /first-time (1ms)                                                                                                                                            nitro 12:24:38
  ├─ /hot (1ms)                                                                                                                                                   nitro 12:24:38
  ├─ /identity (1ms)                                                                                                                                              nitro 12:24:38
  ├─ /jobs (1ms)                                                                                                                                                  nitro 12:24:38
  ├─ /partnership (1ms)                                                                                                                                           nitro 12:24:38
  ├─ /qrCode (0ms)                                                                                                                                                nitro 12:24:38
  ├─ /sales (1ms)                                                                                                                                                 nitro 12:24:38
  ├─ /series-insight (1ms)                                                                                                                                        nitro 12:24:38
  ├─ /settings (2ms)                                                                                                                                              nitro 12:24:38
  ├─ /spotlight (0ms)                                                                                                                                             nitro 12:24:38
  ├─ /sustainability (1ms)                                                                                                                                        nitro 12:24:38
  ├─ /teleport (1ms)                                                                                                                                              nitro 12:24:38
  ├─ /teleport-bridge (0ms)                                                                                                                                       nitro 12:24:38
  ├─ /transfer (1ms)                                                                                                                                              nitro 12:24:38
  ├─ /transform (3ms)                                                                                                                                             nitro 12:24:38
  ├─ /tutorials (1ms)                                                                                                                                             nitro 12:24:38
  ├─ /bsx/assets (0ms)                                                                                                                                            nitro 12:24:38
  ├─ /bsx/claim (1ms)                                                                                                                                             nitro 12:24:38
  ├─ /bsx/create (1ms)                                                                                                                                            nitro 12:24:38
  ├─ /bsx/incomingOffers (0ms)                                                                                                                                    nitro 12:24:38
  ├─ /bsx/offers (1ms)                                                                                                                                            nitro 12:24:38
  ├─ /bsx/stats (1ms)                                                                                                                                             nitro 12:24:38
  ├─ /bsx/waifu (0ms)                                                                                                                                             nitro 12:24:38
  ├─ /ksm/create (1ms)                                                                                                                                            nitro 12:24:38
  ├─ /rmrk/admin (0ms)                                                                                                                                            nitro 12:24:38
  ├─ /rmrk/create (1ms)                                                                                                                                           nitro 12:24:38
  ├─ /rmrk/credit (1ms)                                                                                                                                           nitro 12:24:39
  ├─ /rmrk/mint (1ms)                                                                                                                                             nitro 12:24:39
  ├─ /snek/assets (1ms)                                                                                                                                           nitro 12:24:39
  ├─ /snek/claim (1ms)                                                                                                                                            nitro 12:24:39
  ├─ /snek/create (0ms)                                                                                                                                           nitro 12:24:39
  ├─ /snek/incomingOffers (2ms)                                                                                                                                   nitro 12:24:39
  ├─ /snek/offers (1ms)                                                                                                                                           nitro 12:24:39
  ├─ /snek/stats (1ms)                                                                                                                                            nitro 12:24:39
  ├─ /snek/waifu (0ms)                                                                                                                                            nitro 12:24:39
  ├─ /stmn/create/collection (1ms)                                                                                                                                nitro 12:24:39
  ├─ /stmn/create/item (0ms)                                                                                                                                      nitro 12:24:39
✔ You can now deploy dist to any static hosting!
```

**Comment by vikiival** [comment]
> prepare: [error] Cannot find module '/home/runner/work/nft-gallery/nft-gallery/node_modules/@kodadot1/static/dist/index.cjs'

Loooks like **nft-gallery** is two times there, but I do not know how it appeared there

**Comment by vikiival** [comment]
continue in #5935

**Pull request closed without merging**